_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,213 @@
+
+### Conclusions
+
+Our benchmark experiments highlight the following advantages and trade-offs of using z-score normalization compared to min-max normalization in hybrid search:
+
+- **Search quality (measured using NDCG@10 across four datasets)**:
+
+  * Z-score normalization shows a modest improvement in search quality, with an average increase of 2.08% in NDCG@10 scores.
+  * This suggests that z-score normalization may provide slightly better search result relevance compared to the default min-max normalization.
+
+
+- **Latency impact**:
+
+  * Z-score normalization shows a small increase in latency across different percentiles, as shown in the following table.
+
+  |Latency percentile	|Percent difference	|
+  |---	|---	|
+  |p50	|0.70%	|
+  |p90	|0.50%	|
+  |p99	|0.50%	|
+
+  * The positive percentages indicate that z-score normalization has slightly higher latency compared to min-max normalization, but the differences are minimal (less than 1% on average).
+
+- **Trade-offs**:
+
+  * There's a slight trade-off between search quality and latency. Z-score normalization offers an improvement in search relevance (2.08% increase in NDCG@10) at the cost of a marginal increase in latency (0.50% to 0.72% across different percentiles).
+
+- **Overall assessment**:
+
+  * Z-score normalization provides a modest improvement in search quality with a negligible impact on latency.
+  * The choice between z-score and min-max normalization may depend on your use case. If search relevance is a priority, z-score normalization may be the better option, especially if your system can tolerate a small latency increase.
+
+## Benchmarking Z Score for Production Scenario
+
+Benchmark experiments were conducted using an OpenSearch cluster consisting of a single r6g.8xlarge instance as the coordinator node, along with three r6g.8xlarge instances as data nodes with 12 shards.
+
+### NDCG@mean
```

Comment:
Please add an intro sentence for the table (similar to the one in the section above beginning with "The following table.."